server/github.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/andygrunwald/go-jira"
+	"github.com/google/go-github/v25/github"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+func httpGithubEvent(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
```

Comment:
We are getting the mm user through `.Header.Get("Mattermost-User-Id")`. This is usually used for requests coming from the frontend or MM API.

I assume this is supposed to be the PR submitter. I think we should instead use a different header like `PR-Submitter-MM-User-ID`. Something that expresses what the user id means.

I think we should also have a secret shared between the two plugins, to know that the request is coming from the Github plugin specifically.

Reply:
I agree on the header being more descriptive.

On the secret part, we could send the Github secret the let the Jira plugin verify it (for this purpose I guess the Jira plugin must access the Github config)

Reply:
Follow up on this: we could use SourcePluginId field on the ServeHTTP context, but I'm afraid this is not available on version 5.12 of mattermost-server. I don't know what is the process for upgrading the server version on plugins

Reply:
@enolal826 I thought about this some more, and I think it's important to focus more on the feature, and less on what is going to connect plugins properly. The functionality pertaining specifically to Github and Jira is the core of this feature.

What do you think about leaving the security concerns out of scope for the first iteration of development? The Jira plugin will trust any request coming in through that endpoint for now. It will be simple to fill in the security decisions at any time.